test/tftp/tftp_server_test.rb

```diff
@@ -0,0 +1,110 @@
+require 'test_helper'
+require 'tftp/server'
+require 'tftp/tftp_plugin'
+
+module TftpGenericServerSuite
+  def setup
+    @rootdir = "/some/root"
+    @mac = "aa:bb:cc:dd:ee:ff"
+    @content = "file content"
+    FileUtils.stubs(:mkdir_p).returns(true)
+    Proxy::TFTP::Plugin.settings.stubs(:tftproot).returns(@rootdir)
+    setup_paths
+  end
+
+  def pxe_config_files
+    @pxe_config_files.collect{|f| File.join(@rootdir, f)}
+  end
+
+  def pxe_default_files
+    @pxe_default_files.collect{|f| File.join(@rootdir, f)}
+  end
+
+  def test_set
+    pxe_config_files.each do |file|
+      @subject.stubs(:write_file).with(file, @content)
+    end
+    @subject.expects(:write_file).with(pxe_config_files.first, @content).once
+    @subject.set @mac, @content
+  end
+
+  def test_del
+    pxe_config_files.each do |file|
+      File.stubs(:exist?).with(file).returns(true)
+      FileUtils.stubs(:rm_f).with(file)
+    end
+    @subject.del @mac
+  end
+
+  def test_get
+    file = pxe_config_files.first
+    File.stubs(:exist?).with(file).returns(true)
+    @subject.expects(:read_file).with(file).returns(@content)
+    assert_equal @content, @subject.get(@mac)
+  end
+
+  def test_create_default
+    FileUtils.stubs(:mkdir_p).returns(true)
+    pxe_default_files.each do |file|
+      @subject.stubs(:write_file).with(file, @content)
+    end
+    @subject.expects(:write_file).with(pxe_default_files.first, @content)
```

Comment:
Same issue as I pointed out in the comment for https://github.com/theforeman/smart-proxy/pull/414/files#diff-a25bd017a8ff9ea98b4d47183ffa2d50R23.